flo-runner/src/main/java/com/spotify/flo/context/FloRunner.java

```diff
@@ -57,16 +57,56 @@
 /**
  * This class implements a top-level runner for {@link Task}s.
  */
-public final class FloRunner<T> {
+public class FloRunner<T> {
 
   private static final Logger LOG = LoggerFactory.getLogger(FloRunner.class);
+  private static final TerminationHookFactory DEFAULT_TERMINATION_HOOK_FACTORY = () ->
```

Comment:
It doesn't feel we need a factory. A default hook does nothing but exists should be sufficient.